torch/testing/_internal/common_device_type.py

```diff
@@ -286,8 +301,10 @@ def instantiated_test(self, name=name, test=test_fn, dtype=dtype, op=op):
                 # Sets precision and runs test
                 # Note: precision is reset after the test is run
                 guard_precision = self.precision
+                guard_rel_tol = self.rel_tol
                 try:
                     self.precision = self._get_precision_override(test_fn, dtype)
```

Comment:
I'm a little confused by this line. First `self.precision` is set using `self._get_precision_override`, but then it's set again using `self._get_tolerance_override`?

Reply:
```
245 def _get_tolerance_override(self, test, dtype):
246     if not hasattr(test, 'tolerance_overrides'):
247         return self.precision, self.rel_tol
248     return test.tolerance_overrides.get(dtype, (self.precision, self.rel_tol))

817 # Specifies per-dtype tolerance overrides tol(atol, rtol). It has priority over
818 # precisionOverride.
```

Hi @mruberry, there is a comment that `toleranceOverride` has priority over `precisionOverride`. Unless `toleranceOverride` is defined and it has the override for the corresponding `dtype`, `self.precision` remains unchanged from `_get_precision_override`. Then its value was set from previous `precisionOverride`.

Reply:
Once `toleranceOverride` is in, I think we should just deprecate precisionOverride?

Reply:
I'm still confused about this. The lines are:

```
self.precision = self._get_precision_override(test_fn, dtype)
self.precision, self.rel_tol = self._get_tolerance_override(test_fn, dtype)
```

Now let's inline the functions:

```
self.precision = self.precision if not hasattr(test_fn, 'precision_overrides') else test.precision_overrides.get(dtype, self.precision)
self.precision = self.precision if not hasattr(test_fn, 'tolerance_overrides') else test.tolerance_overrides.get(dtype, (self.precision, self.rel_tol)[0]
self.rel_tol = self.rel_tol if not hasattr(test_fn, 'tolerance_overrides') else test.tolerance_overrides.get(dtype, (self.precision, self.rel_tol)[1]
```

Does that seem correct?

So isn't the value of `self.precision` overwritten by the second line?

Reply:
`self.precision` does get overwritten when `tolerance_overrides` contains the corresponding `dtype`.

Consider the following test cases for `dtype = float`:
```
import torch
from torch.testing._internal.common_device_type import \
    (precisionOverride, toleranceOverride, tol)

class test_obj(object):
    def run(self, test_fn):
        self.precision = 0
        self.rel_tol = 0
        dtype = torch.float

        print(f'=== {test_fn.__name__} ===')
        print(f'{self.precision}, {self.rel_tol}')

        self.precision = self.precision if not hasattr(test_fn, 'precision_overrides') \
            else test_fn.precision_overrides.get(dtype, self.precision)
        print(f'Applied Precision Override: {self.precision}, {self.rel_tol}')

        self.precision = self.precision if not hasattr(test_fn, 'tolerance_overrides') \
            else test_fn.tolerance_overrides.get(dtype, (self.precision, self.rel_tol))[0]
        self.rel_tol = self.rel_tol if not hasattr(test_fn, 'tolerance_overrides') \
            else test_fn.tolerance_overrides.get(dtype, (self.precision, self.rel_tol))[1]
        print(f'Applied Tolerance Override: {self.precision}, {self.rel_tol}')


def test_fn():
    None

@toleranceOverride({torch.float: tol(atol=2.3, rtol=3.4), })
def test_fn_toleranceoverride():
    None

@precisionOverride({torch.float: 1.2})
def test_fn_precisionoverride():
    None

@precisionOverride({torch.float: 1.2})
@toleranceOverride({torch.float: tol(atol=2.3, rtol=3.4), })
def test_fn_float_and_float():
    None

@precisionOverride({torch.float: 1.2})
@toleranceOverride({torch.double: tol(atol=2.3, rtol=3.4), })
def test_fn_float_and_double():
    None

test = test_obj()
test.run(test_fn)
test.run(test_fn_toleranceoverride)
test.run(test_fn_precisionoverride)
test.run(test_fn_float_and_float)
test.run(test_fn_float_and_double)
```

and the results are
```
=== test_fn ===
0, 0
Applied Precision Override: 0, 0
Applied Tolerance Override: 0, 0
=== test_fn_toleranceoverride ===
0, 0
Applied Precision Override: 0, 0
Applied Tolerance Override: 2.3, 3.4
=== test_fn_precisionoverride ===
0, 0
Applied Precision Override: 1.2, 0
Applied Tolerance Override: 1.2, 0
=== test_fn_float_and_float ===
0, 0
Applied Precision Override: 1.2, 0
Applied Tolerance Override: 2.3, 3.4
=== test_fn_float_and_double ===
0, 0
Applied Precision Override: 1.2, 0
Applied Tolerance Override: 1.2, 0
```

Reply:
Aha! You're completely correct, of course, and I should have noticed that from my own inlining. Thank you for elaborating